wys/api/python/wys_google_sheet.py

```diff
@@ -83,12 +83,56 @@ def pull_from_sheet(con, service, dict_table, ward, *args):
         else:
             LOGGER.info('%s does not have any row with different date format', table_name)
 
-    insert = sql.SQL('''INSERT INTO {}.{} (ward_no, location, from_street, to_street, direction, installation_date, removal_date,
-                       new_sign_number, comments, confirmed) VALUES %s 
-                       ON CONFLICT (installation_date, new_sign_number) DO UPDATE SET 
-                       removal_date=EXCLUDED.removal_date, new_sign_number=EXCLUDED.new_sign_number, comments=EXCLUDED.comments
-                       ''').format(sql.Identifier(schema_name), sql.Identifier(table_name)) 
-    LOGGER.info('Uploaded %s rows to PostgreSQL for %s', len(rows), table_name)
+    insert = sql.SQL('''
+        WITH new_data (ward_no, location, from_street, to_street, direction, 
+                    installation_date, removal_date, new_sign_number, comments, 
+                    confirmed) 
+                    AS (
+                VALUES %s) 
+        , dupes AS(
+            /*Identify rows that will violate the unique constraint in the 
+            upsert*/
```

Comment:
This could be more explicit: Identify as duplicates rows where the same sign is installed on the same date and insert into a `duplicates` table